### PR TITLE
Add UART register enumerators

### PR DESCRIPTION
**Pull request by Resaec**
Add UART register enumerators for better access to the following registers

- VENDOR_STATUS_0_0
- LSR_0
- LCR_0
- IIR_FCR_0

Editing uart_wait_idle() to take `UartVendorStatus` as a parameter

### CODE REVIEW

File: exosphere/src/uart.h [hunk 1]
```diff
@@ -22,12 +24,112 @@ typedef enum {
...
+
+    /* This bit is set to 1 when write data is issued to the TX FIFO when it is already full and gets cleared on register read (sticky bit until read)
+    0 = NO_OVERRUN
+    1 = OVERRUN
+    */
+	UART_VENDOR_STATE_TX_OVERRUN = 1 << 3,
+
+	UART_VENDOR_STATE_RX_FIFO_COUNTER = 0b111111 << 16, /* reflects number of current entries in RX FIFO */
+	UART_VENDOR_STATE_TX_FIFO_COUNTER = 0b111111 << 24 /* reflects number of current entries in TX FIFO */
+} UartVendorStatus;
```
Comment (fincs) [inline exosphere/src/uart.h]: This enum seems to have indentation tabs instead of spaces.

Comment (GGLinnk) [reply exosphere/src/uart.h]: Right

Comment (Resaec) [reply exosphere/src/uart.h]: Thanks for pointing that out

**Comment by hexkyz** [comment]
These changes respect the code style and provide clearer definitions for the registers. Merging.

**Pull request merged**